src/handlers/gestures/simpleGestures.ts

```diff
@@ -0,0 +1,429 @@
+import { Gesture } from './gesture';
+import { GestureBuilder, BuiltGesture } from './gestureBuilder';
+import { Directions } from '../../Directions';
+import {
+  GestureEventPayload,
+  HandlerStateChangeEventPayload,
+} from '../gestureHandlerCommon';
+import { getNextHandlerTag } from '../handlersRegistry';
+import { PanGestureHandlerEventPayload } from '../PanGestureHandler';
+import { ForceTouchGestureHandlerEventPayload } from '../ForceTouchGestureHandler';
+import { PinchGestureHandlerEventPayload } from '../PinchGestureHandler';
+import { RotationGestureHandlerEventPayload } from '../RotationGestureHandler';
+
+type HitSlopType =
+  | number
+  | Partial<
+      Record<
+        'left' | 'right' | 'top' | 'bottom' | 'vertical' | 'horizontal',
+        number
+      >
+    >
+  | Record<'width' | 'left', number>
+  | Record<'width' | 'right', number>
+  | Record<'height' | 'top', number>
+  | Record<'height' | 'bottom', number>;
+
+type CommonGestureConfig = {
+  ref: React.RefObject<SimpleGesture>;
+  enabled: boolean;
+  minPointers: number;
+  shouldCancelWhenOutside: boolean;
+  hitSlop: HitSlopType;
+  requireToFail: (number | SimpleGesture | React.RefObject<SimpleGesture>)[];
+  simultaneousWith: (number | SimpleGesture | React.RefObject<SimpleGesture>)[];
+};
+
+type PanGestureHandlerEvent = GestureEventPayload &
+  PanGestureHandlerEventPayload;
+
+type PinchGestureHandlerEvent = GestureEventPayload &
+  PinchGestureHandlerEventPayload;
+
+type RotationGestureHandlerEvent = GestureEventPayload &
+  RotationGestureHandlerEventPayload;
+
+type ForceTouchGestureHandlerEvent = GestureEventPayload &
+  ForceTouchGestureHandlerEventPayload;
+
+type HandlerCallbacks = {
+  handlerTag: number;
+  onBegan?: (event: HandlerStateChangeEventPayload) => void;
+  onStart?: (event: HandlerStateChangeEventPayload) => void;
+  onEnd?: (event: HandlerStateChangeEventPayload, success: boolean) => void;
+  onUpdate?:
+    | ((event: PanGestureHandlerEvent) => void)
+    | ((event: PinchGestureHandlerEvent) => void)
+    | ((event: RotationGestureHandlerEvent) => void)
+    | ((event: ForceTouchGestureHandlerEvent) => void);
+};
+
+export abstract class SimpleGesture extends Gesture {
+  public handlerTag = -1;
+  public handlerName = '';
+  public config: Partial<CommonGestureConfig> = {};
+  public handlers: HandlerCallbacks = {
+    handlerTag: -1,
+  };
+
+  private addDependency(
+    key: 'simultaneousWith' | 'requireToFail',
+    gesture: SimpleGesture | React.RefObject<SimpleGesture>
+  ) {
+    this.config[key] = this.config[key]
+      ? [].concat(this.config[key], gesture)
+      : [gesture];
+  }
+
+  private toArray(x: unknown) {
+    return [].concat(x);
+  }
+
+  setRef(ref: React.RefObject<SimpleGesture>) {
+    this.config.ref = ref;
+    return this;
+  }
+
+  setOnBegan(callback: (event: HandlerStateChangeEventPayload) => void) {
```

Comment:
We could parametrize those with generic param. Check `HandlerStateChangeEvent` type or create your own here.

Reply:
Done in 712e64fc3e4ab072ac817af3700ba4918d277b12